### PR TITLE
fix(fake-browser): ensure nested objects are not mutated on set operation

### PR DESCRIPTION
**Pull request by Timeraa**
This PR fixes a nasty bug that I came across when writing storage tests inside of wxt and I couldn't figure out why at the time and now my friend @bas950 came across the same issue and it turns out that the `fakebrowser` package doesn't remove the references when updating the object.

I tried using `structuredClone` but it didn't quite fix the issue in some deeply nested objects / arrays.

Idk if you remember this @aklinker1 https://discord.com/channels/1212416027611365476/1224505556346212515/1308235414582071358 but this is the fix of that issue.

The bun lock probably got updated because I have the latest version of it. 

---
This pull request includes changes to the `fake-browser` package to improve the handling of nested objects in the storage API. The most important changes include adding a new test case to ensure nested objects are not mutated and modifying the `set` method to deep clone the items before setting them.

Improvements to storage API:

* [`packages/fake-browser/src/apis/storage.test.ts`](diffhunk://#diff-d5fcf764b55e2e1d0f7824b82f8cef338603077250626022dca585b8750cca99R123-R132): Added a test case to ensure that setting a nested object does not mutate the original object.

Codebase modification:

* [`packages/fake-browser/src/apis/storage.ts`](diffhunk://#diff-598be10f09981f99b325c56c13fccd2af7f9bfa249ee4ad6206efd3f55f32bbcL71-R71): Modified the `set` method in `defineStorageArea` to deep clone the items using `JSON.parse(JSON.stringify(items))` before setting them.

### CODE REVIEW

File: packages/fake-browser/src/apis/storage.test.ts [hunk 1]
```diff
@@ -120,4 +120,14 @@ describe('Fake Storage API', () => {
...
   });
+
+  it('setting a nested object should not mutate the original object', async () => {
+    const a = { a: { a: 1, b: [{ c: 1 }] } };
+    await fakeBrowser.storage.local.set(a);
+    const b = await fakeBrowser.storage.local.get();
+    b.a.a = 2;
+    b.a.b[0].c = 2;
+    expect(a.a.a === 1).toBe(true);
+    expect(a.a.b[0].c === 1).toBe(true);
```
Comment (Bas950) [inline packages/fake-browser/src/apis/storage.test.ts:131]:
```suggestion
    expect(a.a.a).toBe(1);
    expect(a.a.b[0].c).toBe(1);
```

**Review by aklinker1 — APPROVED** [review]
Good find! Are there any other APIs that need this as well?

**Comment by Timeraa** [comment]
> Good find! Are there any other APIs that need this as well?
> 

I am not quite sure, I've so far only really used the storage of fake browser

**Comment by aklinker1** [comment]
Now that I know messaging is weird for recursive objects, should probably implement that behavior as well. Not in this PR though, I'll do it separately

**Pull request merged**